garage/tf/models/gaussian_mlp_regressor_model.py

```diff
@@ -0,0 +1,95 @@
+"""GaussianMLPRegressorModel."""
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.models import GaussianMLPModel
+
+
+class GaussianMLPRegressorModel(GaussianMLPModel):
+    """
+    GaussianMLPRegressorModel.
+
+    Args:
+    :param output_dim: Output dimension of the model.
+    :param name: Name of the model.
+    :param hidden_sizes: List of sizes for the fully-connected hidden
+        layers.
+    :param learn_std: Is std trainable.
+    :param init_std: Initial value for std.
+    :param adaptive_std: Is std a neural network. If False, it will be a
+        parameter.
+    :param std_share_network: Boolean for whether mean and std share the same
+        network.
+    :param std_hidden_sizes: List of sizes for the fully-connected layers
+        for std.
+    :param min_std: Whether to make sure that the std is at least some
+        threshold value, to avoid numerical issues.
+    :param max_std: Whether to make sure that the std is at most some
+        threshold value, to avoid numerical issues.
+    :param std_hidden_nonlinearity: Nonlinearity for each hidden layer in
+        the std network.
+    :param std_output_nonlinearity: Nonlinearity for output layer in
+        the std network.
+    :param std_parametrization: How the std should be parametrized. There
+        are a few options:
+        - exp: the logarithm of the std will be stored, and applied a
+            exponential transformation
+        - softplus: the std will be computed as log(1+exp(x))
+    :param hidden_nonlinearity: Nonlinearity used for each hidden layer.
+    :param output_nonlinearity: Nonlinearity for the output layer.
+    """
+
+    def __init__(self, output_dim, name='GaussianMLPRegressorModel', **kwargs):
+        # Network parameters
+        super().__init__(output_dim=output_dim, name=name, **kwargs)
+
+    def network_output_spec(self):
+        """Network output spec."""
+        return [
+            'sample', 'means', 'log_stds', 'std_param', 'normalized_means',
+            'normalized_log_stds', 'x_mean', 'x_std', 'y_mean', 'y_std', 'dist'
+        ]
+
+    def _build(self, state_input):
+
+        input_shape = tuple(state_input.get_shape().as_list()[1:])
```

Comment:
perhaps we need a input_spec as a constructor argument?

Reply:
i think that is a good pattern to use, since policies all have an environment spec, other primitives should have an input/output spec.

that might require you to refactor a lot of code, though, so you can skip it this time (and file an issue instead) if you want to focus on getting this merged.